### PR TITLE
[OTel] Refactor couchbase instrumentation

### PR DESCRIPTION
**Pull request by CagriYonca**
Coverage: 96%

### CODE REVIEW

File: src/instana/instrumentation/couchbase_inst.py [hunk 1]
```diff
@@ -6,90 +6,148 @@
...
+        "observe_multi",
+        "endure_multi",
+        "remove_multi",
+        "counter_multi",
+    ]
+
+    def collect_attributes(
+        span: InstanaSpan,
+        instance: Bucket,
+        query_arg: Any,
```
Comment (pvital) [inline src/instana/instrumentation/couchbase_inst.py:73]: Is there something better than `Any`?

File: src/instana/instrumentation/couchbase_inst.py [hunk 2]
```diff
@@ -6,90 +6,148 @@
...
-            scope.span.set_tag('couchbase.bucket', instance.bucket)
-            scope.span.set_tag('couchbase.type', op)
+            span.set_attribute("couchbase.hostname", instance.server_nodes[0])
+            span.set_attribute("couchbase.bucket", instance.bucket)
+            span.set_attribute("couchbase.type", op)
 
-            if query_arg is not None:
+            if query_arg:
                 query = None
                 if type(query_arg) is N1QLQuery:
```
Comment (pvital) [inline src/instana/instrumentation/couchbase_inst.py:83]: Look, you have a hint for `query_arg`'s type here!

File: src/instana/instrumentation/couchbase_inst.py [hunk 3]
```diff
@@ -6,90 +6,148 @@
...
 
-    def make_wrapper(op):
-        def wrapper(wrapped, instance, args, kwargs):
+    def make_wrapper(op: str) -> Callable:
+        def wrapper(
+            wrapped: Callable[..., object],
+            instance: couchbase.bucket.Bucket,
+            args: Tuple[object, ...],
+            kwargs: Dict[str, Any],
+        ) -> Any:
```
Comment (pvital) [inline src/instana/instrumentation/couchbase_inst.py:99]:
```suggestion
        ) -> object:
```

File: src/instana/instrumentation/couchbase_inst.py [hunk 4]
```diff
@@ -6,90 +6,148 @@
...
 
-            with tracer.start_active_span("couchbase", child_of=parent_span) as scope:
-                capture_kvs(scope, instance, None, op)
+            with tracer.start_as_current_span(
+                "couchbase", span_context=parent_context
+            ) as span:
+                collect_attributes(span, instance, None, op)
                 try:
                     return wrapped(*args, **kwargs)
                 except Exception as e:
```
Comment (pvital) [inline src/instana/instrumentation/couchbase_inst.py:113]: Use the following expression along your file (and all other instrumentation) and skip this block for coverage counting:
```suggestion
                except Exception as exc:
```

File: src/instana/instrumentation/couchbase_inst.py [hunk 5]
```diff
@@ -6,90 +6,148 @@
...
+
         return wrapper
 
-    def query_with_instana(wrapped, instance, args, kwargs):
+    def query_with_instana(
+        wrapped: Callable[..., object],
+        instance: couchbase.bucket.Bucket,
+        args: Tuple[object, ...],
+        kwargs: Dict[str, Any],
+    ):
```
Comment (pvital) [inline src/instana/instrumentation/couchbase_inst.py:125]: Missing return type annotation.

**Review by pvital — CHANGES_REQUESTED** [review]
Few suggestions.

File: src/instana/instrumentation/couchbase_inst.py (hunk 1 above)
Comment (pvital) [reply src/instana/instrumentation/couchbase_inst.py:73]: If there is no way to check which other type is coming, then I prefer `Union[N1QLQuery, object]`.

File: src/instana/instrumentation/couchbase_inst.py [hunk 6]
```diff
@@ -6,90 +6,148 @@
...
+    from couchbase.bucket import Bucket
+
+    if not (
+        hasattr(couchbase, "__version__")
+        and couchbase.__version__[0] == "2"
+        and (
+            couchbase.__version__[2] > "3"
+            or (couchbase.__version__[2] == "3" and couchbase.__version__[4] >= "4")
+        )
+    ):
```
Comment (GSVarsha) [inline src/instana/instrumentation/couchbase_inst.py:28]: Can we make it simpler?

```suggestion
if not hasattr(couchbase, "__version__") and (couchbase.__version__ < "2.3.4" or couchbase.__version__ >= "3.0.0"):
```

**Review by GSVarsha — CHANGES_REQUESTED** [review]
Small suggestion.

**Review by GSVarsha — APPROVED** [review]
Looks good to me!

**Pull request merged**